pkg/cli/admin/release/info.go

```diff
@@ -775,6 +775,28 @@ func (o *InfoOptions) LoadReleaseInfo(image string, retrieveImages bool) (*Relea
 				errs = append(errs, err)
 				return true, nil
 			}
+			// try to verify user-passed image first, in case of mirrored images this will
+			// exist, and if so, use this instead of the release image-reference. If this doesn't
+			// exist, proceed with release image-reference from the readReleaseImageReference above
+			newRef := ref.Ref.AsRepository()
```

Comment:
Assuming that the release image and the target share the same repository is a really ugly hack.  I don't think it's appropriate without the user explicitly telling us where it is (perhaps via a flag, perhaps by providing an ICSP).

Reply:
So is this "pull down any ICSPs from the cluster and teach the local `oc` image resolvers how to process them"?  That would be great, but seems like a lot of work without vendoring more of `containers/image`.  Or does the native `oc` image-resolution code already support mirrors and we'd just need to feed the ICSPs in?

Reply:
... and there is no guarantee that ICSPs that make sense for the cluster also make sense for wherever the user is running `oc`.  Would be nice if we had image-streams for every image referenced from the release image or some such, so we could pull from there without having to reach out and hit some external registry.

Reply:
so ICS is printed when mirroring image like so:
```
imageContentSources:
- mirrors:
  - quay.io/sallyom/release-test
  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
- mirrors:
  - quay.io/sallyom/release-test
  source: registry.svc.ci.openshift.org/ocp/release
```
so if an image is mirrored pass a flag to provide that info? i don't have details for how to use this info atm, i'll take a look.

Reply:
OK, that's fair and w/o that information it's ok to not to have properly working disconnected mirrors working, exactly like described in the bug. The second question is I can't seem to find which piece of `containers/image` is responsible for that, can you point to it?

Reply:
@smarterclayton I'm confused by your comment, `Assuming that the release image and the target share the same repository`.  I don't see where this assumes the same repository.  You mean the shas aren't necessarily the same, if you mirror from one registry to another?

Reply:
(also, as an aside, the mirror that produced that ICS above isn't possible without this PR: https://github.com/openshift/oc/pull/408 as it requires a pull-secret for the quay.io/openshift-release-dev separate than the push-creds for quay.io/sallyom)

Reply:
I just mirrored to a docker registry and it succeeds w/ this PR, fails from master:
```console
first the mirror (requires https://github.com/openshift/oc/pull/408):
$ oc adm release mirror --pull-secret ~/installer/pull-secret registry.svc.ci.openshift.org/ocp/release:4.5.0-0.nightly-2020-05-05-205255 --to docker.io/sallyom/release-test

then extract: (this PR):
$ oc adm release extract --command openshift-install docker.io/sallyom/release-test:4.5.0-0.nightly-2020-05-05-205255
$ ./openshift-install version
./openshift-install 4.5.0-0.nightly-2020-05-05-205255
built from commit abefc8a4e5cb3b8a947722384ddd3c6dfb7fb72d
release image docker.io/sallyom/release-test@sha256:a8b4208250700a544149124bfdbcdc2382490ced704f21ac8e93afd51333ba80

extract (from master)
$ oc adm release extract --command openshift-install docker.io/sallyom/release-test:4.5.0-0.nightly-2020-05-05-205255
error: unable to connect to image repository quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a455c2ffb2e34d951fe8e12c56800a00f4c35420643237c509f1e36350502f98: Get "https://quay.io/v2/": EOF
```

Reply:
It would be more appropriate right now to add a flag that is "--mirror-repository=foo/bar" (possibly that supports multiple flags via StringSlice) than to add magic lookup.  In the future, reeading from an ICSP file or ICSP on the cluster might be reasonable.  Right now, don't do magic.

Reply:
Adding error path when image can't be found to suggest that if this is a mirrored repository to provide that flag, and to add a paragraph to help that describes how to work with mirrored repositories, would also be required to make this clear to users.

Reply:
but how will that solve 'I can't `oc adm release extract` in disconnected environment'? 
As/is, the cmd does the right thing and extracts from the mirrored repository (you don't have to pass it), and succeeds if you are connected and have the correct pull secret for that repo but fails otherwise.

Reply:
That flag would add additional search locations (like ICSP) if the first fetch fails.  All locations that look at an image would support the alternates.

Reply:
You would be designing in (vs adding hacks) a mechanism that says "if i get an image, if i can't get it at location A, try B, C, and D by digest".  That mechanism *probably* should be in the lowest level client (the registry fetcher) which would transparently support this for all commands (i.e. putting code into mirror is probably wrong).